### PR TITLE
ci: run trivy on source directory

### PR DESCRIPTION
**Pull request by jedevc**
This should hopefully improve visibility of new vulnerabilities.

cc @gerhard @samalba

### CODE REVIEW

File: ci/engine.go
Comment (jedevc) [inline ci/engine.go:303]: @TomChv any ideas about this? :cry: 

I genuinely cannot seem to find a way to get this into a good state right now - it seems like the `ws` dependency is vulnerable - and `websocket-stream` (somewhere in our dep tree) uses a horrendously old version that hasn't got a backport (and of course, looks like `websocket-stream` is abandoned).

IMO I think it's totally reasonable to ignore vulnerabilities in the non-dagger example code, but I wonder if we would want to rework our examples so we don't have this problem in the first place.

Comment (jedevc) [reply ci/engine.go:303]: FYI, this same issue is affecting the `docs` directory.

Comment (jedevc) [reply ci/engine.go]: I've expanded this selection to exclude `docs` for now.

Not that those aren't issues, but IMO the `engine scan` pipeline should scan for issues in the engine/sdks/etc - not necessarily the examples and docs, and such.

**Review by gerhard — APPROVED** [review]
This got lost in my backlog.

Approving optimistically. OK to merge if we rebase & all checks pass 👍

File: ci/engine.go
Comment (TomChv) [reply ci/engine.go]: That's a good point! I agree that we should change our example to not use outdated dependencies
@vikram-dagger, could you work on this if you have some time?

**Review by TomChv — APPROVED** [review]
Ok to merge too after rebase

**Comment by jedevc** [comment]
Scan is now succeeding after fixing the java dep, and rebasing, so merging now :tada:

**Pull request merged**
